tianshou/policy/modelfree/bcq.py

```diff
@@ -0,0 +1,124 @@
+import torch
+import torch.nn.functional as F
+import numpy as np
+from typing import Any, Dict, Union, Optional
+
+from tianshou.data import Batch
+from tianshou.policy import BasePolicy
+from copy import deepcopy
+
+
+class BCQPolicy(BasePolicy):
+    """Implementation discrete BCQ algorithm. Some code is from
+    https://github.com/sfujim/BCQ/tree/master/discrete_BCQ
+
+    """
+
+    def __init__(
+        self,
+        model: torch.nn.Module,
+        optim: torch.optim.Optimizer,
+        tau: float,
+        target_update_frequency: int,
+        device: str,
+        gamma: float,
+        imitation_logits_penalty: float,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(**kwargs)
+        self._policy_net = model
+        self._policy_net.to(device)
```

Comment:
I suggest that `to(device)` should not stay at the policy level. Either in the test code `net.to(args.device)` or `policy.to(args.device)`, or pass this `device` argument into the network (BCQN below) like the other examples in `utils/net`, so that you don't need to care about the device in `BCQPolicy`. That would make the code cleaner.